test/functional/feature_bind_port_discover.py

```diff
@@ -52,28 +78,34 @@ def skip_test_if_missing_module(self):
 
     def run_test(self):
         self.log.info(
-                "Test that if -bind= is not passed then all addresses are "
+                "Test that if -bind= is not passed or -bind=0.0.0.0 is used then all addresses are "
                 "added to localaddresses")
-        found_addr1 = False
-        found_addr2 = False
-        for local in self.nodes[0].getnetworkinfo()['localaddresses']:
-            if local['address'] == ADDR1:
-                found_addr1 = True
-                assert_equal(local['port'], BIND_PORT)
-            if local['address'] == ADDR2:
-                found_addr2 = True
-                assert_equal(local['port'], BIND_PORT)
-        assert found_addr1
-        assert found_addr2
+        for i in [0, 1, 2]:
+            found_addr1 = False
+            found_addr2 = False
+            for local in self.nodes[i].getnetworkinfo()['localaddresses']:
+                if local['address'] == ADDR1:
+                    found_addr1 = True
+                    assert_equal(local['port'], self.bind_ports[i])
+                if local['address'] == ADDR2:
+                    found_addr2 = True
+                    assert_equal(local['port'], self.bind_ports[i])
+            if not found_addr1:
+                self.log.error(f"Address {ADDR1} not found in node{i}'s local addresses: {local}")
+                assert False
+            if not found_addr2:
+                self.log.error(f"Address {ADDR2} not found in node{i}'s local addresses: {local}")
+                assert False
```

Comment:
The variable `local` in these lines is being used outside the `for` loop which defines it. If `localaddresses` is empty `[]` the error message itself will cause an error.